tokio/src/sync/notify.rs

```diff
@@ -130,6 +130,7 @@ enum NotificationType {
 }
 
 #[derive(Debug)]
+#[repr(C)] // required by `linked_list::Link` impl
```

Comment:
nit: this should probably be mentioned in the `# Safety` comment for the `Link` impl, as it's required for that impl to be sound?